File: src/Java/ExampleCode/Test2.java
package Java.ExampleCode;

public class Test2 {

    public static void main(String[] args) {
        B b = new B(2);
    }
}

class A {
    public A() {
        System.out.println("생성자 A");
    }

    int a;

    public A(int a) {
        this.a = a;
        System.out.println("생성자 A2");
    }
}

class B extends A {
    public B() {
        System.out.println("생성자 B");
    }

    int b;
    public B(int b) {
        super(b);
        this.b = b;
        System.out.println("생성자 B2");
    }
}

File: src/ExampleCode/Test2.kt
package ExampleCode

/**
 * Test2
 *
 * @author JungGyun.Choi
 * @version 1.0.0
 * @since 2024. 11. 11.
 */
fun main() {
    val list = listOf(
        Auth("b", false),
        Auth("c", false),
        Auth("d", false),
        Auth("a", true),
    )

    val result = list.any {it.b }
    println("result: $result")
}

data class Auth(
    val a: String,
    val b: Boolean
)